jsonschema-core.xml

```diff
@@ -135,13 +135,66 @@
                 data in several ways.
             </t>
             <t>
-                JSON Schema uses keywords to assert constraints on JSON instances or annotate those
-                instances with additional information.  Additional keywords are used to apply
-                assertions and annotations to more complex JSON data structures, or based on
-                some sort of condition.
+                A JSON Schema document is comprised of a set of keywords,
+                which are specified as properties in a JSON object.
+                The name of the keyword is used as the property name,
+                and any arguments to the keyword are provided as the value.
+                Keywords can be classified by their functions.
+                The most important classes are validation, annotation, and applicator keywords.
             </t>
             <t>
-                To facilitate re-use, keywords can be organized into vocabularies. A vocabulary
+                "Validation keywords" form the basis of JSON Schema.
+                Given a schema and an instance, the instance is "valid" if all of the keywords in
+                the schema are valid against the instance.
+                Validation keywords may be used to describe a set of JSON documents.
+                Without any validation keywords, the set of all valid instances is the set of all valid JSON documents.
+                Typically, adding validation keywords creates a subset.
+                Keywords are typically defined to not be redundant with other keywords:
+                For example, the "minimum" keyword only shrinks the valid set of numbers, but not strings or other
+                types&#x2014;this is what the "type" keyword does.
+            </t>
+            <t>
+                When an instance validates against a schema, "annotation keywords" may provide "annotations":
```

Comment:
Regardless of the above (whether validation is more important), do we call the process of applying annotations "validation"? I think it's just an independent process one can do even without performing validation, no?

Reply:
The language emphasizes that annotations are a byproduct of validation (an invalid instance doesn't produce annotations). Mostly, I haven't introduced the term "applying" yet, and I don't think "validation" is _wrong_.

Reply:
I am saying I don't think that that's how we talk about annotations previously. One can "process" an instance, not validate anything, and collect some annotations as far as I know, and we support and encourage that use case.

Reply:
The production of annotations requires validation because no annotations are returned if validation fails.

Conversely many validation keywords require annotation output from other keywords.

They are inextricably linked.

Reply:
So I can't define a keyword with annotation result without saying it participates in some validation process? I guess I could have answered that by looking at what we say for title or description or default. I stand corrected then, especially since handrews in another comment also said we do indeed call the general process validation.

Reply:
If your schema has no assertions (and you don't use applicators that convert a passing subschema to a validation failure, e.g. `not` and `oneOf`) it can't fail validation, therefore validation is only involved to a trivial degree.  And as I noted before, yes, I have used JSON Schema that way (sadly the project is not public).

Reply:
Looking back again I'm honestly confused now by the bottom of this :D -- @handrews (I know you're not fully responsive so feel free to delay here obviously) -- you +1'ed my initial comment, but then Greg contradicted it pointing out it was wrong, and then your last comment seems ambiguous :D so I again am confused as to whether I'm wrong or not, so you'll forgive me for asking it again:

What do we call the process itself of applying (where I'm not using a technical term here by "applying") the schema `{"title" : "Cool Stuff", "default": 12}` to an instance `"foo"` in an implementation with no support whatsoever for the validation vocabulary, it's some completely different use of JSON Schema as it sounds like the project you're referencing may have done.

Do we still call that process "validation", just now in a vacuous sense?

Reply:
I think "validation" is currently used, but I like the more generic "evaluation."  "Apply" or "application" may be construed as related to "applicator" keywords specifically.

Reply:
I'd go with "evaluation."  I think "apply" is also fine whether the application is because of an applicator or because of a top-level invocation, but to avoid confusion, "evaluate" is probably best (and I think we settled on `evaluationPath` in the revised output format?)

Whether or not you can have annotation without validation depends on how you conceptualize it:

1. "Annotations happen unless validation fails"
2. "Annotations only happen if validation succeeds"

A schema that does not contain any keywords that can produce a validation failure by definition cannot fail validation.  If you think of that as performing validation that succeeds, and only then having annotations, then technically you always have validation.

If you think of validation as a process that only occurs when it's possible to fail (meaning that something like `title` wouldn't really be performing validation because it can't cause a failure), then a schema with no keywords that can fail isn't doing validation but it does produce annotations.

I conceptualize it as option 1:  the schema `{"title": "foo"}` produces an annotation.  Since there are no assertions, and no applicators that can fail validation in the absence of assertions, then there is no validation going on that would drop those annotations.

But you can think of it the other way around and it works just as well.

And as @gregsdennis said, you have some assertions that require** annotations to function correctly, so you could also say that annotation is just as fundamental.  Or you could say that if none of your keywords produce annotations, then no annotation is happening, just validation.

Which I think proves that neither should be framed as the primary behavior.

All of this is addressed in _a lot_ more detail in the presentation that I should work on rather than answering all of this stuff, but answering short comments is easier right now.

**there's some wording to allow people to do something different in terms of actual code, but conceptually they require annotations